### ISSUE TITLE
MLOps 

### ISSUE THREAD

**Issue by YunchaoYang** [YunchaoYang/Blogs#27]
# Outline
This post will elaborate on how to effectively use Machine Learning infrastructure. Namely, the best practice in building, maintaining and scaling production-ready deep learning systems. 

0) Build a production ready deep learning pipeline
   
   Tensoflow extended (TFX)
   pytorch ?
   Nvidia ? 

1) Kubernetes with Google Cloud: Deploy your Deep Learning model effortlessly.

2) Scalability in ML

3) Docker containers and Docker Compost 

4) uWSGI Nginx
serving a Tensorflow model to users with Flask, uWSGI as a web server and Nginx as a reverse proxy. 

5) Deploy a Deep Learning model as a web application with Flask
 

# references: 
1. https://theaisummer.com/topics/mlops/

**Comment by YunchaoYang** [comment]
# Tutorials on using MLflow on HPG

1. Open an OOD terminal
2. Load all required modules
<code>  module load conda firefox git </code>
3. create an conda environment with MLflow installed

**Comment by YunchaoYang** [comment]
The MLOps projects are roughly categorized into the following:
1. Training Orchestration
2. Model Monitoring
3. Model Testing
4. Model Serving
5. Data Versioning
6. Feature Store
7. Experiment Tracking
8. Explainability

## Another different categorization strategy:

1. Automates ML workflow
2. CI/CD for ML
3. cron job scheduler:
   Tools for monitoring cron jobs, a command line job scheduler on Unix like operating system
4. Data Catalog
5. Data Enrichment
6. Data Exploration
7. Data Management
8. Data Processing
9. Data Validation
10. Data Visualization
11. Feature Engineering
12. Feature Store
13. Hyperparameter Tuning
14. Knowledge Sharing
15. Machine Learning Platform
16. and so on

**Comment by YunchaoYang** [comment]
develop ML products and rapidly bring them into production. 

automate and operationalize ML products

concept drift/data drift
